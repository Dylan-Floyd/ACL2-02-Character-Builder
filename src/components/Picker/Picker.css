.picker {
  display: flex;
  flex-direction: column;
}

.picker > * {
  margin: 5px
}

label > * {
  margin: 0px 5px;
}