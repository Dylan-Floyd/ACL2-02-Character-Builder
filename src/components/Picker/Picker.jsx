import React from 'react'

import './Picker.css'

export default function Picker({
  head,
  setHead,
  torso,
  setTorso,
  legs,
  setLegs,
  addCatchphrase,
  newCatchphrase,
  setNewCatchphrase
}) {
  return (
    <div className='picker'>
      <label>
        Head:
        <select value={head} onChange={(e) => setHead(e.target.value)}>
          <option value="bird">Bird</option>
          <option value="dog">Dog</option>
          <option value="duck">Duck</option>
          <option value="horse">Horse</option>
        </select>
      </label>
      <label>
        Torso:
        <select value={torso} onChange={(e) => setTorso(e.target.value)}>
          <option value="blue">Blue</option>
          <option value="dress">Dress</option>
          <option value="pink">Pink</option>
          <option value="red">Red</option>
        </select>
      </label>
      <label>
        Legs:
        <select value={legs} onChange={(e) => setLegs(e.target.value)}>
          <option value="blue">Blue</option>
          <option value="leg">Leg</option>
          <option value="white">White</option>
        </select>
      </label>
      <label>
        Add a Catchphrase:
        <input type="text" value={newCatchphrase} onChange={(e) => setNewCatchphrase(e.target.value)}></input>
        <button onClick={addCatchphrase}>Add</button>
      </label>
    </div>
  )
}
