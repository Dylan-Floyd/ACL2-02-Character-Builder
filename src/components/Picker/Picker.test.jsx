import { screen, render } from '@testing-library/react'
import Picker from './Picker.jsx'

it('should render our Picker component', () => {
  const { container } = render(<Picker />)
  const head = screen.getByLabelText(/Head/)
  const middle = screen.getByLabelText(/Torso/)
  const legs = screen.getByLabelText(/Legs/)
  const catchphrase = screen.getByLabelText(/Add a Catchphrase/)

  expect(head).toBeInTheDocument()
  expect(middle).toBeInTheDocument()
  expect(legs).toBeInTheDocument()
  expect(catchphrase).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})