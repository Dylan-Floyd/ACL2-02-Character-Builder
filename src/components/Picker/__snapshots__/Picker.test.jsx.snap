// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render our Picker component 1`] = `
<div>
  <div
    class="picker"
  >
    <label>
      Head:
      <select>
        <option
          value="bird"
        >
          Bird
        </option>
        <option
          value="dog"
        >
          Dog
        </option>
        <option
          value="duck"
        >
          Duck
        </option>
        <option
          value="horse"
        >
          Horse
        </option>
      </select>
    </label>
    <label>
      Torso:
      <select>
        <option
          value="blue"
        >
          Blue
        </option>
        <option
          value="dress"
        >
          Dress
        </option>
        <option
          value="pink"
        >
          Pink
        </option>
        <option
          value="red"
        >
          Red
        </option>
      </select>
    </label>
    <label>
      Legs:
      <select>
        <option
          value="blue"
        >
          Blue
        </option>
        <option
          value="leg"
        >
          Leg
        </option>
        <option
          value="white"
        >
          White
        </option>
      </select>
    </label>
    <label>
      Add a Catchphrase:
      <input
        type="text"
        value=""
      />
      <button>
        Add
      </button>
    </label>
  </div>
</div>
`;
